src/ui/components/SecondaryToolbox/TerminalContextAdapter.tsx

```diff
@@ -20,16 +20,15 @@ export default function TerminalContextController({ children }: PropsWithChildre
   const idCounterRef = useRef(0);
 
   const addMessage = useCallback((partialTerminalExpression: NewTerminalExpression) => {
+    const terminalExpression: TerminalExpression = {
```

Comment:
nit: Can you ad dan inline comment of why we're doing this? It looks like the kind of thing that might get refactored back to how it was before if someone doesn't know.

I think the only/main thing you're trying to do is prevent the `idCounterRef.current` value from being incremented more than once?

Reply:
> I think the only/main thing you're trying to do is prevent the idCounterRef.current value from being incremented more than once?

That was my initial concern but now I've added some more logic to ensure that the same message isn't added multiple times if the callback to `setMessages()` is called multiple times. I'm not sure if that extra logic is needed but I added it just in case. I've also added a warning that the callback may be called multiple times.